ttnn/tensor.py

```diff
@@ -400,31 +400,23 @@ def requires_padding_change(layout, shape):
             return False
 
     necessary_to_change_padding = requires_padding_change(layout, tensor.shape)
-    if not necessary_to_change_padding and not layout_change_needed:
-        return tensor
+
```

Comment:
are you sure we don't need this short circuit on the check here?  I seem to recall we needed it.

Reply:
I added an early return if `not layout_change_needed` in an earlier PR.

So, if we are here, we need to change the layout